pkg/kubelet/util/pod_startup_latency_tracker_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/component-base/metrics/testutil"
+	"k8s.io/kubernetes/pkg/kubelet/metrics"
+	testingclock "k8s.io/utils/clock/testing"
+)
+
+var frozenTime = time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC)
+
+const (
+	uid         = "3c1df8a9-11a8-4791-aeae-184c18cca686"
+	metricsName = "kubelet_pod_start_sli_duration_seconds"
+)
+
+func TestSinglePod(t *testing.T) {
```

Comment:
We need more tests, e.g.:
* the one that observes again pod for recorded=true state
* the one for observing pod which isn't in a desired version.
* Maybe more?
If there is any bug in such logic (or someone introduces any with code change), we want to detect it here instead of when debugging production clusters.

Reply:
Makes sense - I've added two more tests.